src/hooks/useAllV3Ticks.ts

```diff
@@ -0,0 +1,108 @@
+import { Token } from '@uniswap/sdk-core'
+import { FeeAmount, TICK_SPACINGS } from '@uniswap/v3-sdk'
+import { nearestUsableTick, TickMath } from '@uniswap/v3-sdk/dist/'
+import { ZERO_ADDRESS } from '../constants/misc'
+import { useEffect, useMemo, useState } from 'react'
+import { Result, useSingleCallResult, useSingleContractMultipleData } from 'state/multicall/hooks'
+import { useTickLens, useV3Factory } from './useContract'
+import { TickData } from 'constants/ticks'
+
+function bitmapIndex(tick: number, tickSpacing: number) {
+  return Math.floor(tick / tickSpacing / 256)
+}
+
+const REFRESH_FREQUENCY = { blocksPerFetch: 24 }
+
+// for now, reconsider using this function, it consumes a lot of data and cpu to fetch all the ticks.
+export function useAllV3Ticks(
+  token0: Token | undefined,
+  token1: Token | undefined,
+  feeAmount: FeeAmount | undefined
+): {
+  loading: boolean
+  syncing: boolean
+  error: boolean
+  valid: boolean
+  tickData: TickData[]
+} {
+  const tickSpacing = feeAmount && TICK_SPACINGS[feeAmount]
+
+  const minIndex = useMemo(
+    () => (tickSpacing ? bitmapIndex(nearestUsableTick(TickMath.MIN_TICK, tickSpacing), tickSpacing) : undefined),
+    [tickSpacing]
+  )
+  const maxIndex = useMemo(
+    () => (tickSpacing ? bitmapIndex(nearestUsableTick(TickMath.MAX_TICK, tickSpacing), tickSpacing) : undefined),
+    [tickSpacing]
+  )
+
+  const [tickDataLatestSynced, setTickDataLatestSynced] = useState<TickData[]>([])
+
+  // fetch the pool address
+  const factoryContract = useV3Factory()
+  const poolAddress = useSingleCallResult(factoryContract, 'getPool', [token0?.address, token1?.address, feeAmount])
+    .result?.[0]
+
+  const tickLensArgs: [string, number][] = useMemo(
+    () =>
+      maxIndex && minIndex && poolAddress && poolAddress !== ZERO_ADDRESS
+        ? new Array(maxIndex - minIndex + 1)
+            .fill(0)
+            .map((_, i) => i + minIndex)
+            .map((wordIndex) => [poolAddress, wordIndex])
+        : [],
+    [minIndex, maxIndex, poolAddress]
+  )
+
+  const tickLens = useTickLens()
+  const callStates = useSingleContractMultipleData(
+    tickLensArgs.length > 0 ? tickLens : undefined,
+    'getPopulatedTicksInWord',
+    tickLensArgs,
+    REFRESH_FREQUENCY,
+    3_000_000
+  )
+
+  const error = useMemo(() => callStates.some(({ error }) => error), [callStates])
+  const loading = useMemo(() => callStates.some(({ loading }) => loading), [callStates])
+  const syncing = useMemo(() => callStates.some(({ syncing }) => syncing), [callStates])
+  const valid = useMemo(() => callStates.some(({ valid }) => valid), [callStates])
+
+  const tickData = useMemo(
```

Comment:
let's move this inside the `useEffect` conditional, it should save some cycles, and as long as `error`/`loading`/`syncing`/`valid` are referentially constant (which they should be), the behavior should be the same?